bridge-token-factory/src/lib.rs

```diff
@@ -342,6 +413,145 @@ impl BridgeTokenFactory {
             );
     }
 
+    pub fn get_deposit_token_fee(&self, token: &String) -> Option<DepositFee> {
+        self.deposit_fee.get(token)
+    }
+
+    pub fn get_withdraw_token_fee(&self, token: &String) -> Option<WithdrawFee> {
+        self.withdraw_fee.get(token)
+    }
+
+    // Returns fee percentage for different tokens per silo
+    pub fn get_withdraw_fee_percentage_per_silo(
+        &self,
+        silo_account_id: &AccountId,
+        token: &String,
+    ) -> Option<U128> {
+        self.withdraw_fee_percentage_per_silo
+            .get(&get_silo_fee_map_key(&silo_account_id, Some(token)))
+            .or_else(|| {
+                self.withdraw_fee_percentage_per_silo
+                    .get(&get_silo_fee_map_key(&silo_account_id, None))
+            })
+    }
+
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_deposit_fee(
+        &mut self,
+        token: &String,
```

Comment:
We need to convert the token address to lowercase, otherwise, it will be possible to pass the valid uppercase address

Reply:
13509deca41650d8e1c89c95f4dd10525e7a6fe4